FirebaseInstallations/Source/Library/InstallationsAPI/FIRInstallationsItem+RegisterInstallationAPI.m

```diff
@@ -73,15 +63,22 @@ @implementation FIRInstallationsItem (RegisterInstallationAPI)
   return installation;
 }
 
-- (NSString *)validStringOrNilForKey:(NSString *)key fromDict:(NSDictionary *)dict {
-  NSString *string = dict[key];
-  if ([string isKindOfClass:[NSString class]] && string.length > 0) {
-    return string;
+#pragma mark - Auth token
+
++ (nullable FIRInstallationsStoredAuthToken *)authTokenWithGenerateTokenAPIJSONData:(NSData *)data
+                                                                               date:(NSDate *)date
+                                                                              error:(NSError **)
+                                                                                        outError {
+  NSDictionary *dict = [self dictionaryFromJSONData:data error:outError];
```

Comment:
jUst double check is it going to crash if data is nil?

Reply:
Good question. Yes, it will. I don't think it is possible for NSURLSessionDataTask to finish with a response and data == nil, but I added handling of it just in case.